iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp

```diff
@@ -47,7 +47,7 @@ class PoshRuntimeSingleProcess_test : public Test
 
 TEST_F(PoshRuntimeSingleProcess_test, ConstructorPoshRuntimeSingleProcessIsSuccess)
 {
-    iox::RouDiConfig_t defaultRouDiConfig = iox::RouDiConfig_t().setDefaults();
+    iox::RouDiConfig_t defaultRouDiConfig = iox::RouDiConfig_t().optimize();
```

Comment:
this was changed as to improve code coverage for https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_posh/source/roudi/roudi_config.cpp

Reply:
> this was changed as to improve code coverage for https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_posh/source/roudi/roudi_config.cpp

It is dangerous to increase the unittest coverage for a class which is not tested here because this can lead to wrong assumptions. RouDiConfig is not the test subject here and should be tested in a separate testsuite.
The `optimize()` method is not intended to be used for creating a default roudi config.
https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

Reply:
okay i will revert this changes, could you please take a look in rest of the test?

Reply:
reverted changes

Reply:
for roudi_config.hpp i am try to add below test, 
TEST(RoudiConfig_test, OptimizeIsSucessfull)
{
    iox::config::RouDiConfig sut;
    auto optimizedRoudiConfig = sut.optimize();
    EXPECT_EQ(sut, optimizedRoudiConfig);
}

this requires to add operator= in struct RouDiConfig(). 

@dkroenke  Is a good way to continue?